db/sqlglue.c

```diff
@@ -791,6 +794,8 @@ void done_sql_thread(void)
         destroy_sqlite_master(thd->rootpages, thd->rootpage_nentries);
         free(thd);
     }
+
+    bdb_temp_table_maybe_reset_priority_thread(thedb->bdb_env);
```

Comment:
You should do this IFF the thread has requested temptables. For queries that don't need a temp table at all (eg, SELECT * FROM TABLE), this is needless. This is why this is done in bdb_temp_table_close().